### PR TITLE
GO-4472 try to use api

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: tantivy_test.go [hunk 1]
```diff
@@ -498,6 +499,76 @@ func Test(t *testing.T) {
...
+		err = tc.AddAndConsumeDocuments(doc, doc2)
+		require.NoError(t, err)
+
+		docs, err := tc.NumDocs()
+		require.NoError(t, err)
+		require.Equal(t, uint64(2), docs)
+
+		qb := tantivy_go.NewQueryBuilder()
+
+		finalQuery := qb.
```
Comment (fat-fellow) [inline tantivy_test.go:522]: This is example how it looks like

File: tantivy_test.go [hunk 2]
```diff
@@ -515,29 +514,37 @@ func Test(t *testing.T) {
...
+			Query(tantivy_go.Should, "title2", "term2", tantivy_go.PhrasePrefixQuery, 0.1).
+			Query(tantivy_go.MustNot, "title3", "term2", tantivy_go.SingleTermPrefixQuery, 0.1).
+			BooleanQuery(tantivy_go.Must, tantivy_go.NewQueryBuilder().
+				Query(tantivy_go.Should, "summary", "term3", tantivy_go.PhrasePrefixQuery, 1.0).
+				BooleanQuery(tantivy_go.Should, tantivy_go.NewQueryBuilder().
+					Query(tantivy_go.Must, "comments", "not single term", tantivy_go.PhraseQuery, 0.8),
+				),
+			).
+			Build()
+
```
Comment (fat-fellow) [inline tantivy_test.go:535]: Here I use the data json

File: rust/src/queries/convert.rs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+    use std::fs;
+    use tantivy::query::PhrasePrefixQuery;
+    use tantivy::schema::{IndexRecordOption, Schema, TextFieldIndexing, STORED, TEXT};
+    use tantivy::tokenizer::{
+        AsciiFoldingFilter, Language, LowerCaser, RemoveLongFilter, SimpleTokenizer, Stemmer,
+        TextAnalyzer,
+    };
+    use tantivy::Index;
+
+    fn expected_query() -> FinalQuery {
```
Comment (fat-fellow) [inline rust/src/queries/convert.rs:156]: Here the query from jsom

File: rust/src/queries/convert.rs [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+        let index = Index::create_in_ram(schema.clone());
+        index.tokenizers().register("simple", text_analyzer_simple);
+
+        let parsed = convert_to_tantivy(&index, given_query, &schema).expect("can't convert");
+
+        let expected = expected_tantivy_query_str();
+
+        assert_eq!(expected, format!("{parsed:#?}"));
+    }
+
```
Comment (fat-fellow) [inline rust/src/queries/convert.rs:290]: Here the validation

**Review by zarkone — APPROVED** [review]

**Pull request merged**